core/server/src/test/java/alluxio/master/file/PermissionCheckerTest.java

```diff
@@ -194,6 +194,14 @@ public void before() throws Exception {
     mPermissionChecker = new PermissionChecker(sTree);
   }
 
+  // Helper function to create a path and set the permission to what specified in option.
+  private static void createAndSetPermission(String path, CreateFileOptions option)
+      throws Exception {
+    InodeTree.CreatePathResult result = sTree.createPath(new AlluxioURI(path), option);
+    result.getCreated().get(result.getCreated().size() - 1)
```

Comment:
I find the problem is that the PermissionStatus object bundles two things that are to be treated differently: 1) user name and group name, which need to be set based on different context (e.g. login module or thrift transport) and 2) permission bits, which do not need to be set

The only way I see fixing this is to get rid of PermissionStatus on the code path that creates and Inode and only (if at all) use it to describe the permission status of an inode. In other words, the CreatePathOptions would have user and group fields (but not permission bits). When inode is created, the user and group is taken from the options and the permission bits are constructed using the defaults + umask.